src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/entity/NameEntity.kt

```diff
@@ -41,25 +47,32 @@ class AliasEntity(
   @Column(name = "date_of_birth")
   val dateOfBirth: LocalDate? = null,
 
+  @Column
+  @Enumerated(STRING)
+  val type: NameType,
+
   @Version
   var version: Int = 0,
 ) {
   companion object {
-    private fun from(alias: Alias): AliasEntity? =
+    private fun from(name: Name): NameEntity? =
       when {
-        isAliasPresent(alias.firstName, alias.middleNames, alias.lastName) ->
-          AliasEntity(
-            firstName = alias.firstName,
-            middleNames = alias.middleNames,
-            lastName = alias.lastName,
+        isAliasPresent(name.title, name.firstName, name.middleNames, name.lastName) ->
+          NameEntity(
+            title = name.title,
+            firstName = name.firstName,
+            middleNames = name.middleNames,
+            lastName = name.lastName,
+            dateOfBirth = name.dateOfBirth,
+            type = name.type,
           )
         else -> null
       }
 
-    fun fromList(aliases: List<Alias>): List<AliasEntity> = aliases.mapNotNull { from(it) }
+    fun fromList(names: List<Name>): List<NameEntity> = names.mapNotNull { from(it) }
 
-    private fun isAliasPresent(firstName: String?, middleNames: String?, surname: String?): Boolean =
-      sequenceOf(firstName, middleNames, surname)
+    private fun isAliasPresent(title: String?, firstName: String?, middleNames: String?, surname: String?): Boolean =
+      sequenceOf(title, firstName, middleNames, surname)
```

Comment:
Not sure this adds much - we would not want a `Name` which consisted only of a `title` would we?

Reply:
Done 👍